fera_selection_utilities.py

```
@@ -0,0 +1,150 @@
+    closest_span2 = None
+    closest_span2cont = None
+    closest_span2page = None
+    min_distance = float('inf')
+    min_distance2 = float('inf')
+    ponto1 = Point([mousepos1.x, mousepos1.y])
+    ponto2 = Point([mousepos2.x, mousepos2.y])
+    inispan = [None, -1]
+    endspan = [None, -1]
+    cont = 0
+    for page in range(pageinitsel, pageendsel + 1):
+        for block in self.pixmaps[page][1]['blocks']:
```

Ruff (0.15.10)

[error] Undefined name `self`

(F821)